src/forwarder.ts

```diff
@@ -63,6 +79,43 @@ async function validateForwarderArn(aws: Aws, functionArn: CloudFormationObjectA
   }
 }
 
+export async function addExecutionLogGroupsAndSubscriptions(
+  service: Service,
+  aws: Aws,
+  functionArn: CloudFormationObjectArn | string,
+) {
+  const extendedProvider = (service.provider as any)?.logs;
+
+  if (!isLogsConfig(extendedProvider)) {
+    return;
+  }
+
+  const resources = service.provider.compiledCloudFormationTemplate?.Resources;
+  if (restExecutionLoggingIsEnabled(extendedProvider)) {
+    // create log group
+    const logGroupName = await createRestExecutionLogGroupName(aws);
+    const executionLogGroupKey = "RestExecutionLogGroup";
+    const executionLogGroupName = addExecutionLogGroup(logGroupName);
+    resources[executionLogGroupKey] = executionLogGroupName;
+    // add subscription
+    const executionSubscription = subscribeToExecutionLogGroup(functionArn, executionLogGroupKey);
+    const executionSubscriptionKey = "RestExecutionLogGroupSubscription";
+    resources[executionSubscriptionKey] = executionSubscription;
```

Comment:
Maybe I am missing something here -- it looks like the `resources` object here is shared across all of the functions. If we set the key to a single subscription, then run the function again, will we overwrite the subscription from the first time we ran the function?

Reply:
I don't think so. It might rewrite it in the `serverless_state.json` but it shouldn't have an effect in Cloudwatch. I don't know that there is a way to avoid this based on how the hooks work.

Reply:
OK -- if there is a separate `resources` object created for each time we call this function then it should work. If you've tested this with multiple functions and it correctly subscribes to all of their logs then this is fine :+1:

Reply:
I tested this a bit more and it looks OK.